### PR TITLE
Start marking functions that can transitively trigger a GC

### PR DESCRIPTION
**Pull request by jdm**
Per #33140, adding this zero-size argument to functions that can transitively GC makes it easier to recognize where hazards are present. This work can be done incrementally and iteratively as time allows.

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes are part of #33140.
- [x] These changes do not require tests because there is no functional difference.

### CODE REVIEW

**Comment by jdm** [comment]
This PR gets us to a point where I can file followups for less experienced contributors to help.oush it forward. There are many threads to pull!

File: components/script/dom/bindings/reflector.rs [hunk 1]
```diff
@@ -25,20 +25,29 @@ where
...
+            CanGc::note(),
+        )
+    }
 }
 
 pub fn reflect_dom_object_with_proto<T, U>(
     obj: Box<T>,
     global: &U,
     proto: Option<HandleObject>,
+    can_gc: CanGc,
```
Comment (gterzian) [inline components/script/dom/bindings/reflector.rs:43]: is there a future need to pass it here, as opposed to instantiate it inside the function and hide it from the caller?

**Review by gterzian — APPROVED** [review]
LGTM, with one question.

File: components/script/dom/bindings/reflector.rs (hunk 1 above)
Comment (jdm) [reply components/script/dom/bindings/reflector.rs:43]: Yes. The purpose is to infect callers so it is clear whether any function can trigger a GC.

**Comment by jdm** [comment]
@last-genius Good question! A useful model for adding arguments to the generated methods is the `InRealm` arguments, which are manually annotated in Bindings.conf. there is code in CodegenRust.py that checks for these annotations and adds the argument if they're present.

**Pull request merged**